%
%
% UCSD Doctoral Dissertation Template
% -----------------------------------
% http://ucsd-thesis.googlecode.com
%
%


%% REQUIRED FIELDS -- Replace with the values appropriate to you

% No symbols, formulas, superscripts, or Greek letters are allowed
% in your title.
\title{Computational analysis of single-cell alternative splicing}

\author{Olga Borisovna Botvinnik}
\degreeyear{\the\year}

% Master's Degree theses will NOT be formatted properly with this file.
\degreetitle{Doctor of Philosophy}

\field{Bioinformatics and Systems Biology}
%\specialization{Anthropogeny}  % If you have a specialization, add it here

\chair{Professor Gene Yeo}
% Uncomment the next line iff you have a Co-Chair
\cochair{Professor Sheng Zhong}
%
% Or, uncomment the next line iff you have two equal Co-Chairs.
%\cochairs{Professor Chair Masterish}{Professor Chair Masterish}

%  The rest of the committee members  must be alphabetized by last name.
\othermembers{
Professor Amy Pasquinelli\\
Professor C. Titus Brown\\
Professor Kun Zhang\\
Professor Sam Pfaff
}
\numberofmembers{5} % |chair| + |cochair| + |othermembers|


%% START THE FRONTMATTER
%
\begin{frontmatter}

%% TITLE PAGES
%
%  This command generates the title, copyright, and signature pages.
%
\makefrontmatter

%% DEDICATION
%
%  You have three choices here:
%    1. Use the ``dedication'' environment.
%       Put in the text you want, and everything will be formated for
%       you. You'll get a perfectly respectable dedication page.
%
%
%    2. Use the ``mydedication'' environment.  If you don't like the
%       formatting of option 1, use this environment and format things
%       however you wish.
%
%    3. If you don't want a dedication, it's not required.
%
%
\begin{dedication}
  To my family, my parents, and Kwasi.
\end{dedication}


% \begin{mydedication} % You are responsible for formatting here.
%   \vspace{1in}
%   \begin{flushleft}
% 	To me.
%   \end{flushleft}
%
%   \vspace{2in}
%   \begin{center}
% 	And you.
%   \end{center}
%
%   \vspace{2in}
%   \begin{flushright}
% 	Which equals us.
%   \end{flushright}
% \end{mydedication}



%% EPIGRAPH
%
%  The same choices that applied to the dedication apply here.
%
\begin{epigraph} % The style file will position the text for you.
  \emph{You have as many hours in a day as Beyonc\'e}
%   ---Smarty Pants
\end{epigraph}

% \begin{myepigraph} % You position the text yourself.
%   \vfil
%   \begin{center}
%     {\bf Think! It ain't illegal yet.}
%
% 	\emph{---George Clinton}
%   \end{center}
% \end{myepigraph}


%% SETUP THE TABLE OF CONTENTS
%
\tableofcontents
\listoffigures  % Comment if you don't have any figures
\listoftables   % Comment if you don't have any tables



%% ACKNOWLEDGEMENTS
%
%  While technically optional, you probably have someone to thank.
%  Also, a paragraph acknowledging all coauthors and publishers (if
%  you have any) is required in the acknowledgements page and as the
%  last paragraph of text at the end of each respective chapter. See
%  the OGS Formatting Manual for more information.
%
\begin{acknowledgements}
 Thank you to Yan for being a patient mentor.
\end{acknowledgements}


%% VITA
%
%  A brief vita is required in a doctoral thesis. See the OGS
%  Formatting Manual for more information.
%
\begin{vitapage}
\begin{vita}
  \item[2010] S.~B. in Mathematics, Massachusetts Institute of Technology
  \item[2010] S.~B. in Biological Engineering, Massachusetts Institute of Technology
  \item[2010] M.~S. in Biomolecular Engineering and Bioinformatics, University of California, Santa Cruz
  \item[2017] Ph.~D. in Bioinformatics and Systems Biology, University of California, San Diego
\end{vita}

\begin{publications}
\item Yan Song*, \textbf{Olga B Botvinnik}*, Michael T Lovci, Boyko Kakaradov, Patrick Liu, Jia L. Xu and Gene W Yeo. Single-cell alternative splicing analysis with Expedition reveals splicing dynamics during neuron differentiation. \emph{In review}. * These authors contributed equally to this work.
\item Curtis A Nutter, Elizabeth A Jaworski, Sunil K Verma, Vaibhav Deshmukh, Qiongling Wang, \textbf{Olga B Botvinnik}, Mario J Lozano, Ismail J Abass, Talha Ijaz, Allan R Brasier, Nisha J Garg, Xander H T Wehrens, Gene W Yeo, and Muge N Kuyumcu-Martinez. Dysregulation of RBFOX2 Is an Early Event in Cardiac Pathogenesis of Diabetes. \emph{Cell Reports}, 15(10):2200-2213, 2016.
\item Jong Wook Kim*, \textbf{Olga B Botvinnik}*, Omar Abudayyeh, Chet Birger, Joseph Rosen- bluh, Yashaswi Shrestha, Mohamed E Abazeed, Peter S Hammerman, Daniel DiCara, David J Konieczkowski, et al. Characterizing genomic alterations in cancer by complementary functional associations. \emph{Nature Biotechnology, 2016}. * These authors contributed equally to this work.
\item P Compeau and P Pevzner. \emph{Bioinformatics Algorithms} Volume 1, volume 1 of An Active Learning Approach. Active Learning Publishers LLC, 2 edition, 2015. Contributed text, figures, problems and code solutions, primarily to ``Chapter 4: How Do We Sequence Antibiotics?''.
\item Kris C Wood, David J Konieczkowski, Cory M Johannessen, Jesse S Boehm, Pablo Tamayo, \textbf{Olga B Botvinnik}, Jill P Mesirov, William C Hahn, David E Root, Levi A Garraway, et al. MicroSCALE screening reveals genetic modifiers of therapeutic response in melanoma. \emph{Science Signaling}, 5(224):rs4, 2012.
\item A Goncearenco, P Grynberg, Olga B Botvinnik, Geoff Macintyre, and Thomas Abeel. Highlights from the Eighth International Society for Computational Biology (ISCB) Student Council Symposium 2012. \emph{BMC Bioinformatics}, 2012.
\item Naomi Galili, Pablo Tamayo, \textbf{Olga B Botvinnik}, Jill P Mesirov, Margarita R Brooks, Gail Brown, and Azra Raza. Prediction of response to therapy with ezatiostat in lower risk myelodysplastic syndrome. \emph{Journal of Hematology \& Oncology}, 5(1):1, 2012.
\item Naomi Galili, Pablo Tamayo, \textbf{Olga B Botvinnik}, Jill P Mesirov, Jennifer Zikria, Gail Brown, and Azra Raza. Gene Expression Studies May Identify Lower Risk Myelodys- plastic Syndrome Patients Likely to Respond to Therapy with Ezatiostat Hydrochloride
(TLK199). \emph{Blood}, 118(21):2779-2779, 2011.
\item Michael F Berger, Gwenael Badis, Andrew R Gehrke, Shaheynoor Talukder, Anthony A Philippakis, Lourdes Pena-Castillo, Trevis M Alleyne, Sanie Mnaimneh, \textbf{Olga B Botvinnik}, Esther T Chan, et al. Variation in homeodomain DNA binding revealed by high-resolution analysis of sequence preferences. \emph{Cell}, 133(7):1266-1276, 2008.
\end{publications}

% \begin{publications}
%   \item Your Name, ``A Simple Proof Of The Riemann Hypothesis'', \emph{Annals of Math}, 314, 2007.
%   \item Your Name, Euclid, ``There Are Lots Of Prime Numbers'', \emph{Journal of Primes}, 1, 300 B.C.
% \end{publications}
\end{vitapage}


%% ABSTRACT
%
%  Doctoral dissertation abstracts should not exceed 350 words.
%   The abstract may continue to a second page if necessary.
%
\begin{abstract}
  Alternative splicing (AS) generates isoform diversity critical for cellular identity and homeostasis in multicellular life. Although AS variation has been observed among single cells for a few events, little is known about the biological significance of such variation. We developed Expedition, a computational framework consisting of outrigger, a \emph{de novo} splice graph transversal algorithm to detect AS; anchor, a Bayesian approach to assign modalities and bonvoyage, a visualization tool using non-negative matrix factorization to display modality changes. Applying Expedition to single iPSCs undergoing neuronal differentiation, we discover up to 20\% of AS exons exhibit bimodality and are flanked by more conserved introns harboring distinct cis-regulatory motifs. Bimodal exons constitute the majority of cell-type specific splicing, are highly dynamic during cellular transitions, preserve translatability and reveal intricacy of cell states invisible to global gene expression analysis. Systematic AS characterization in single cells redefines our understanding of AS complexity in cell biology.

\end{abstract}


\end{frontmatter}
